.github/workflows/main.yml

```diff
@@ -1,32 +1,77 @@
 name: CI
 
 on:
-  # runs on pushes to the main branch
   push:
     branches: [main]
-  # also runs inside pull requests that target the main branch
   pull_request:
     branches: [main]
 
 jobs:
-  build:
+  install:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
-          node-version: "14"
+          node-version: "18"
           cache: "npm"
-      # install dependencies
       - run: npm ci
-      # run ESLint & Prettier
-      - run: npm run lint
+
+  lint:
+    runs-on: ubuntu-latest
+    needs: install
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+          cache: "npm"
+      - run: npm ci
+      - run: npm run lint:code
+      - run: npm run lint:styles
       - run: npm run prettier
-      # verify that there are no type and build errors
+      - run: npm run check:types
+
+  build:
+    runs-on: ubuntu-latest
+    needs: install
+    env:
+      NEXT_PUBLIC_API_BASE_URL: https://prolog-api.profy.dev
```

Comment:
This shouldn't be necessary anymore. I added this env variable to the repository settings. This should have been done automatically but sometimes it fails unfortunately

![CleanShot 2023-10-24 at 08 50 52@2x](https://github.com/profydev/prolog-app-fvaldi/assets/9722792/d68d26bf-a130-4806-b9b9-339d84e5acbc)

Reply:
Thanks a lot Johannes for the fast response.

Kind regards.